config/avahi/avahi.inc

```diff
@@ -203,9 +223,49 @@ function avahi_sync() {
 	}
 	avahi_write_config();
 	// Is package enabled?
-	if (($config['installedpackages']['avahi']['config'][0]['enable']) && file_exists("/usr/local/etc/rc.d/avahi-daemon.sh")) {
+	if ($config['installedpackages']['avahi']['config'][0]['enable']) {
 		start_service("avahi");
 	}
 }
 
+function avahi_upgrade_config() {
+	global $config, $avahi_config;
+
+	$avahi_config = $config['installedpackages']['avahi']['config'][0];
+	if (!is_array($avahi_config)) {
+		$avahi_config = array();
+	}
+	$changes = 0;
+	// Convert previous disable values to enable ones
+	if ($avahi_config['disable_ipv4'] != "on" ) {
```

Comment:
Same here, check if key isset() before attempt to change it. Also, you probably want to assign $avahi_config =& to be able to change $config values, and unset old keys